vector/src/main/java/im/vector/app/features/voicebroadcast/usecase/StartVoiceBroadcastUseCase.kt

```diff
@@ -53,15 +65,41 @@ class StartVoiceBroadcastUseCase @Inject constructor(
 
     private suspend fun startVoiceBroadcast(room: Room) {
         Timber.d("## StartVoiceBroadcastUseCase: Send new voice broadcast info state event")
-        room.stateService().sendStateEvent(
-                eventType = STATE_ROOM_VOICE_BROADCAST_INFO,
+        val chunkLength = VoiceBroadcastConstants.DEFAULT_CHUNK_LENGTH // Todo Get the length from the room settings
+        val eventId = room.stateService().sendStateEvent(
+                eventType = VoiceBroadcastConstants.STATE_ROOM_VOICE_BROADCAST_INFO,
                 stateKey = session.myUserId,
                 body = MessageVoiceBroadcastInfoContent(
                         voiceBroadcastStateStr = VoiceBroadcastState.STARTED.value,
-                        chunkLength = 5L, // TODO Get length from voice broadcast settings
+                        chunkLength = chunkLength,
                 ).toContent()
         )
 
-        // TODO start recording audio files
+        startRecording(room, eventId, chunkLength)
+    }
+
+    private fun startRecording(room: Room, eventId: String, chunkLength: Int) {
+        voiceBroadcastRecorder?.listener = VoiceBroadcastRecorder.Listener { file ->
+            sendVoiceFile(room, file, eventId)
+        }
+        voiceBroadcastRecorder?.startRecord(room.roomId, chunkLength)
+    }
+
+    private fun sendVoiceFile(room: Room, voiceMessageFile: File, referenceEventId: String) {
+        val outputFileUri = FileProvider.getUriForFile(
+                context,
+                buildMeta.applicationId + ".fileProvider",
+                voiceMessageFile,
+                "Voice message.${voiceMessageFile.extension}"
+        )
+        val audioType = outputFileUri.toMultiPickerAudioType(context) ?: return
+        if (audioType.duration > 1000) {
```

Comment:
Do we want to keep this check on the duration? It was for voice message IIRC.

If we keep it, the else block should delete the file, maybe?

Reply:
Yes, I was hesitating on that point, I think it makes sense to remove it, let's see what's happen 🤞🏻